ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrClusterParticipant.java

```diff
@@ -105,21 +117,44 @@ public HelixVcrClusterParticipant(CloudConfig cloudConfig, ClusterMapConfig clus
    * @param partitionIdStr The partitionIdStr notified by Helix.
    */
   public void addPartition(String partitionIdStr) {
-    PartitionId partitionId = partitionIdMap.get(partitionIdStr);
+    // Add partition with retry logic in case Ambry clustermap is not update to date.
+    retryExecutor.runWithRetries(retryPolicy, callback -> doAddPartition(partitionIdStr, callback), e -> true,
```

Comment:
I don't have to intend to do blocking or async execution. Both work to my understanding on Helix/

Reply:
So it is okay for the doAddPartition work to go on in the background after addPartition returns? I guess we are also swallowing/logging any exception that happens after multiple retries in doAddPartition.

Reply:
So it is okay for the doAddPartition work to go on in the background after addPartition returns? : YES based on my current logic. Let me double check with Helix team.

 I guess we are also swallowing/logging any exception that happens after multiple retries in doAddPartition.
Didn't get this. If all retries failed, we could see the exception log in the user call back, no?

Reply:
Do we have a metric that we can monitor to tell if all partition additions succeeded?

Reply:
It's hard to know `all partitions` because helix assign partition in a streaming style. 

We could add metric to indicate partition add failure count.

Reply:
Oh, here I guess I meant that if there was an existing metric that tells the total number of partitions in the VCR cluster, we can know if the VCR cluster has converged with the server cluster. I guess we can also get this information from the helix admin UI.